### PR TITLE
feat(notifications): add new "info" styled Alert and Notification

### PR DESCRIPTION
**Pull request by jzempel**
* [ ] **BREAKING CHANGE?** 

## Description

Brings over the new styling from `css-callouts`.

## Detail

https://github.com/zendeskgarden/css-components/pull/174

## Checklist

* [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
* [x] :nail_care: view component styling is based on a Garden CSS
  component
* [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
* [x] :arrow_left: renders as expected with reversed (RTL) direction
* [x] :guardsman: includes new unit and snapshot tests
* [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/notifications/src/Alert.js [hunk 1]
```diff
@@ -31,16 +32,17 @@ const Alert = styled(Well).attrs({
...
     })
 })`
   ${props => retrieveTheme(COMPONENT_ID, props)};
 `;
 
 Alert.propTypes = {
-  recessed: PropTypes.bool,
-  floating: PropTypes.bool,
-  type: PropTypes.oneOf([VALIDATION.SUCCESS, VALIDATION.WARNING, VALIDATION.ERROR]).isRequired
+  type: PropTypes.oneOf([VALIDATION.SUCCESS, VALIDATION.WARNING, VALIDATION.ERROR, VALIDATION.INFO])
```
Comment (austin94) [inline packages/notifications/src/Alert.js]: Why did we remove the existing propTypes? It still gets passed through to the Well component.

Comment (jzempel) [reply packages/notifications/src/Alert.js]: They are not valid for the `<Alert>` component – there should never be a `floating` alert (and `recessed` isn't applicable). Note given the styled components extension, this only affects documentation. Same for `recessed` on `<Notification>` below.

Comment (jzempel) [reply packages/notifications/src/Alert.js]: p.s. @allisonacs approved ™️

File: packages/notifications/src/Alert.js [hunk 2]
```diff
@@ -31,16 +32,17 @@ const Alert = styled(Well).attrs({
       // Validation types
       [CalloutStyles['c-callout--success']]: props.type === VALIDATION.SUCCESS,
       [CalloutStyles['c-callout--warning']]: props.type === VALIDATION.WARNING,
-      [CalloutStyles['c-callout--error']]: props.type === VALIDATION.ERROR
+      [CalloutStyles['c-callout--error']]: props.type === VALIDATION.ERROR,
+      [CalloutStyles['c-callout--info']]: props.type === VALIDATION.INFO,
+      [CalloutStyles['c-callout--recessed']]: props.type === VALIDATION.INFO
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/notifications/src/Alert.js]: Good call – done.

File: packages/notifications/src/Alert.js (hunk 1 above)
Comment (austin94) [reply packages/notifications/src/Alert.js]: 👍

**Pull request merged**